### PR TITLE
{vis}[foss/2023b] ParaView v5.12.0 w/ Qt5

### PR DESCRIPTION
**Pull request by SimonPinches**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by SimonPinches** [comment]
Qt6 support in ParaView 5.12.0 still appears to be experimental ([see this paraview PR](https://gitlab.kitware.com/paraview/paraview/-/merge_requests/6134)).  Some users have reported significant performance benefits when using Qt5 with ParaView for some applications - hence this PR.

**Comment by furstj** [comment]
Could you add VTK_PYTHON_OPTIONAL_LINK=OFF to your configuration? It seems to be necessary for OpenFOAM in-situ visualization with Catalyst. See e.g.  https://discourse.paraview.org/t/advice-on-building-paraview-superbuild-for-openfoam-runtimepostprocessing/8707.

**Comment by SimonPinches** [comment]
> for OpenFOAM in-situ visualization with Catalyst.

Done :-)

**Comment by SimonPinches** [comment]
PR https://github.com/easybuilders/easybuild-easyconfigs/pull/21002 fixes the issues addressed by this PR so it's no longer needed.

**Comment by furstj** [comment]
Are you sure that PV wit Qt6 from #21002 solves the "Some users have reported significant performance benefits when using Qt5 with ParaView for ..." ?

**Comment by SimonPinches** [comment]
Maybe not for all users so I can re-open this PR in case it's useful :-)

File: easybuild/easyconfigs/p/ParaView/ParaView-5.12.0-foss-2023b-Qt5.eb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    # EGL, X and Mesa
+    '-DOPENGL_glu_LIBRARY=$EBROOTLIBGLU/lib/libGLU.%s' % SHLIB_EXT,
+    '-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include',
+    '-DEGL_INCLUDE_DIR=$EBROOTLIBGLVND/include',
+    '-DEGL_LIBRARY=$EBROOTLIBGLVND/lib/libEGL.%s' % SHLIB_EXT,
+    '-DEGL_opengl_LIBRARY=$EBROOTLIBGLVND/libOpenGL.%s' % SHLIB_EXT,
+    '-DVTK_OPENGL_HAS_EGL=ON',
+    '-DVTK_USE_X=ON',
+    '-DVTK_OPENGL_HAS_OSMESA=OFF',
+    '-DVTK_PYTHON_OPTIONAL_LINK=OFFVTK_PYTHON_=OFF']
```
Comment (boegel) [inline easybuild/easyconfigs/p/ParaView/ParaView-5.12.0-foss-2023b-Qt5.eb:60]:
```suggestion
    '-DVTK_PYTHON_OPTIONAL_LINK=OFF']
```

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3
CORE_CNT=16

File: easybuild/easyconfigs/p/ParaView/ParaView-5.12.0-foss-2023b-Qt5.eb (hunk 1 above)
Comment (SimonPinches) [reply easybuild/easyconfigs/p/ParaView/ParaView-5.12.0-foss-2023b-Qt5.eb]: Thanks @boegel :-)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3116.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/450dcbdd2ad4e18f2cab88666ccf28ed for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @SimonPinches!

**Pull request merged**